src/NzbDrone.Core/Download/Clients/RQBit/RQbitProxy.cs

```diff
@@ -0,0 +1,234 @@
+        {
+            var endpoint = removeData ? "/delete" : "/forget";
+            var itemRequest = BuildRequest(settings).Resource("/torrents/" + infoHash + endpoint);
+            _httpClient.Post(itemRequest.Build());
+        }
+
+        public string AddTorrentFromUrl(string torrentUrl, RQbitSettings settings)
+        {
+            string infoHash = null;
+            var itemRequest = BuildRequest(settings).Resource("/torrents?overwrite=true").Post().Build();
+            itemRequest.SetContent(torrentUrl);
+            var httpResponse = _httpClient.Post(itemRequest);
+            if (httpResponse.StatusCode != HttpStatusCode.OK)
+            {
+                return infoHash;
+            }
+
+            var response = JsonConvert.DeserializeObject<PostTorrentResponse>(httpResponse.Content);
+
+            if (response.Details != null)
+            {
+                infoHash = response.Details.InfoHash;
+            }
+
+            return infoHash;
+        }
+
+        public string AddTorrentFromFile(string fileName, byte[] fileContent, RQbitSettings settings)
+        {
+            string infoHash = null;
+            var itemRequest = BuildRequest(settings)
+                .Post()
+                .Resource("/torrents?overwrite=true")
+                .Build();
+            itemRequest.SetContent(fileContent);
+            var httpResponse = _httpClient.Post(itemRequest);
+            if (httpResponse.StatusCode != HttpStatusCode.OK)
+            {
+                return infoHash;
+            }
+
+            var response = JsonConvert.DeserializeObject<PostTorrentResponse>(httpResponse.Content);
+
+            if (response.Details != null)
+            {
+                infoHash = response.Details.InfoHash;
+            }
+
+            return infoHash;
+        }
+
+        public void SetTorrentLabel(string hash, string label, RQbitSettings settings)
+        {
+            _logger.Warn("Torrent labels currently unsupported by RQBit");
+        }
+
+        public bool HasHashTorrent(string hash, RQbitSettings settings)
+        {
+            var result = true;
+            var rqBitTorrentResponse = GetTorrent(hash, settings);
+            if (rqBitTorrentResponse == null || string.IsNullOrWhiteSpace(rqBitTorrentResponse.InfoHash))
+            {
+                result = false;
+            }
+
+            return result;
+        }
+
+        private TorrentResponse GetTorrent(string infoHash, RQbitSettings settings)
+        {
+            TorrentResponse result = null;
+            var itemRequest = BuildRequest(settings).Resource("/torrents/" + infoHash);
+            var itemResponse = _httpClient.Get(itemRequest.Build());
+            if (itemResponse.StatusCode != HttpStatusCode.OK)
+            {
+                return result;
+            }
+
+            result = JsonConvert.DeserializeObject<TorrentResponse>(itemResponse.Content);
+
+            return result;
```

Comment:
Similar suggestion here, no need to set the result to immediately return it.

```suggestion
            var itemRequest = BuildRequest(settings).Resource("/torrents/" + infoHash);
            var itemResponse = _httpClient.Get(itemRequest.Build());

            if (itemResponse.StatusCode != HttpStatusCode.OK)
            {
                return null;
            }

            return JsonConvert.DeserializeObject<TorrentResponse>(itemResponse.Content);
```